sdk/src/main/java/com/horizen/secret/PrivateKey25519.java

```diff
@@ -21,13 +22,19 @@ public final class PrivateKey25519 implements Secret
 
     public PrivateKey25519(byte[] privateKeyBytes, byte[] publicKeyBytes)
     {
+        Objects.requireNonNull(privateKeyBytes, "Private key can't be null");
+        Objects.requireNonNull(publicKeyBytes, "Public key can't be null");
+
         if(privateKeyBytes.length != PRIVATE_KEY_LENGTH)
             throw new IllegalArgumentException(String.format("Incorrect private key length, %d expected, %d found", PRIVATE_KEY_LENGTH,
                     privateKeyBytes.length));
         if(publicKeyBytes.length != PUBLIC_KEY_LENGTH)
-            throw new IllegalArgumentException(String.format("Incorrect pubKey length, %d expected, %d found", PUBLIC_KEY_LENGTH,
+            throw new IllegalArgumentException(String.format("Incorrect public key length, %d expected, %d found", PUBLIC_KEY_LENGTH,
                     publicKeyBytes.length));
 
+        if(!Ed25519.validatePublicKey(privateKeyBytes, publicKeyBytes))
+            throw new IllegalArgumentException(String.format("Public key is not valid"));
```

Comment:
why do we validate keys in the constructor for this type of key, but for others we do it during deserialization?

Reply:
let's put validation into the `parse` method.